File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSub = nums[0]
    let currSub = 0

    for(const n of nums){
        if(currSub < 0){
            currSub = 0
        }
        currSub = currSub + n
        maxSub = Math.max(currSub, maxSub)
    }
    return maxSub
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sub = nums[0]
        max_s = nums[0]

        for num in nums[1:]:
            sub = max(sub+num, num)
            max_s = max(sub, max_s)

        return max_s
            

        